src/Microsoft.AspNetCore.Mvc.Razor/Internal/SymbolsUtility.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Internal
+{
+    /// <summary>
+    /// Utility type for determining if a platform supports symbol file generation.
+    /// </summary>
+    public class SymbolsUtility
+    {
+        private const string SymWriterGuid = "0AE2DEB0-F901-478b-BB9F-881EE8066788";
+
+        /// <summary>
+        /// Determines if the current platform supports symbols (pdb) generation.
+        /// </summary>
+        /// <returns><c>true</c> if pdb generation is supported; <c>false</c> otherwise.</returns>
+        public static bool SupportsFullPDBGeneration()
```

Comment:
`SupportsFullPdbGeneration()`; no all-caps words or acronyms in the ASP.NET Core APIs e.g. `HtmlMarkupParser` and `XmlHelpers` in Razor and `HttpContext` in HttpAbstractions.